akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -126,6 +127,40 @@ private[akka] final case class MonitorInterceptor[T](actorRef: ActorRef[T]) exte
 
 }
 
+/**
+ * Log all messages for this decorated ReceiveTarget[T] to logger before receiving it ourselves.
+ *
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final case class LogMessagesInterceptor[T](opts: LogOptions) extends BehaviorInterceptor[T, T] {
+  private val optionalAdapter: Option[LoggingAdapter] = Option(opts.adapter)
+
+  import BehaviorInterceptor._
+
+  override def aroundReceive(ctx: TypedActorContext[T], msg: T, target: ReceiveTarget[T]): Behavior[T] = {
+    if (opts.enabled)
+      optionalAdapter.fold(ctx.asScala.log.log(opts.level, "received message {}", msg)) { adapter ⇒
```

Comment:
@patriknw The type returned by `scaladsl.ActorContext[T].log` doesn't extend `akka.event.LoggingAdapter` (AFAICT), so in the implementation, I have to handle logging differently when using the default `ActorContext[T].log` or an adapter.  Am I missing something?

Reply:
sorry, my suggestion was misleading. You should use `akka.actor.typed.Logger` instead of `akka.event.LoggingAdapter`. That is the new API for logging in typed.

Reply:
Ok, cool.  I switched it to `Logger` instead.